ironic-deployment/certmanager/certificate.yaml

```diff
@@ -0,0 +1,69 @@
+apiVersion: cert-manager.io/v1
+kind: Issuer
+metadata:
+  name: selfsigned-issuer
+  namespace: capm3-system
+spec:
+  selfSigned: {}
+---
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: ironic-cacert
+  namespace: capm3-system
+spec:
+  isCA: true
+  ipAddresses:
+  - 172.22.0.2
+  issuerRef:
+    kind: Issuer
+    name: selfsigned-issuer
+  secretName: ironic-cacert
+---
+apiVersion: cert-manager.io/v1
+kind: Issuer
+metadata:
+  name: ca-issuer
+  namespace: capm3-system
+spec:
+  ca:
+    secretName: ironic-cacert
+---
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: ironic-cert
+  namespace: capm3-system
+spec:
+  ipAddresses:
+  - 172.22.0.2
```

Comment:
I am a bit worried about hardcoding this IP? What might be the consequence if we use parameterization through configmap?

Reply:
Not sure if there is a way for cert-manager to get the IP from a configmap. It will be nice if there is. Otherwise, hardcoding the IP is the only way IMO

Reply:
I was trying to parameterize that variable and seems that ipAddresses in cert-manager is accepting only string value and didn't have a way to use configmap. I do have the same opinion as Nam.

Reply:
ok lets tackle this separately.

Reply:
I am referring to something similar to https://github.com/kubernetes-sigs/cluster-api/blob/master/config/certmanager/certificate.yaml#L19